cdap-api/src/main/java/io/cdap/cdap/api/app/AppStateReader.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.app;
+
+/**
+ * Read saved state for the given app
+ */
+public interface AppStateReader {
+
+  /**
+   * Method for returning the saved state for given app and key.
+   * @param namespace Namespace for the app
+   * @param appName Name of the app
+   * @param key Key for the state
+   * @return value as byte[]
+   * @throws AppStateException if the app is not available or otherwise unable to fetch state
+   */
+  byte[] getState(String namespace, String appName, String key) throws AppStateException;
```

Comment:
Why need to have namespace and appName? It should be only used by the current app of reading/writing state?

Reply:
Done